src/reducers/CompareResults.ts

```diff
@@ -1,44 +1,56 @@
 import { createSlice, PayloadAction } from '@reduxjs/toolkit';
 
-import { fetchCompareResults } from '../thunks/compareResultsThunk';
-import type { CompareResultsState, ResultsHashmap } from '../types/state';
+import {
+  fetchCompareResults,
+  fetchFakeResults,
+} from '../thunks/compareResultsThunk';
+import type {
+  CompareResultsState,
+  CompareResultsItem,
+  ResultsHashmap,
+} from '../types/state';
 
 const initialState: CompareResultsState = {
-  data: {},
+  data: {} as ResultsHashmap,
   loading: false,
   error: undefined,
 };
 
+function fetchPending(state: CompareResultsState) {
+  state.loading = true;
+  state.error = initialState.error;
+}
+
+function fetchFulfilled(
+  state: CompareResultsState,
+  action: PayloadAction<CompareResultsItem[]>,
+) {
+  const revisionHash = action.payload[0].new_rev;
+  state.data[revisionHash] = action.payload;
+  state.loading = initialState.loading;
+}
+
+function fetchRejected(
+  state: CompareResultsState,
+  action: PayloadAction<string | undefined>,
+) {
+  state.error = action.payload;
+  state.loading = initialState.loading;
+}
+
 const compareResults = createSlice({
   name: 'compareResults',
   initialState,
-  reducers: {
-    setCompareData(
-      state,
-      action: PayloadAction<{
-        data: ResultsHashmap;
-      }>,
-    ) {
-      state.data = action.payload.data;
-    },
-  },
+  reducers: {},
   extraReducers: (builder) => {
     builder
-      .addCase(fetchCompareResults.fulfilled, (state, action) => {
-        const revisionHash = action.payload[0].new_rev;
-        state.data[revisionHash] = action.payload;
-        state.loading = initialState.loading;
-      })
-      .addCase(fetchCompareResults.pending, (state) => {
-        state.loading = true;
-        state.error = initialState.error;
-      })
-      .addCase(fetchCompareResults.rejected, (state, action) => {
-        state.error = action.payload;
-        state.loading = initialState.loading;
-      });
+      .addCase(fetchCompareResults.pending, fetchPending)
+      .addCase(fetchFakeResults.pending, fetchPending)
+      .addCase(fetchCompareResults.fulfilled, fetchFulfilled)
+      .addCase(fetchFakeResults.fulfilled, fetchFulfilled)
+      .addCase(fetchCompareResults.rejected, fetchRejected)
+      .addCase(fetchFakeResults.rejected, fetchRejected);
```

Comment:
I tried to use `addMatcher` to specify each reducer only once, but I found that the result were more messy.